pkg/inject/sidecar.go

```diff
@@ -98,6 +103,9 @@ func Select(target *appsv1.Deployment, ns *corev1.Namespace, availableJaegerPods
 		// jaeger instance to use!
 		// first, we make sure we normalize the name:
 		jaeger := &availableJaegerPods.Items[0]
+		if target.Annotations == nil {
```

Comment:
not related to this PR but I got a couple of times `"assignment to entry in nil map" (assignment to entry in nil map)` when testing this feature.